extension/src/test/suite/Formatter_mock.test.ts

```diff
@@ -0,0 +1,392 @@
+import * as chai from "chai";
+import * as fs from "fs";
+import * as path from "path";
+import { beforeEach } from "mocha";
+import { LispFormatter } from "../../format/formatter";
+import { ReadonlyDocument } from "../../project/readOnlyDocument";
+import * as fmtConfig from "../../format/fmtconfig";
+import { ImportMock } from "ts-mock-imports";
+import * as resources from "../../resources";
+
+let assert = chai.assert;
+let testDir = path.join(__dirname + "/../../../extension/src/test");
+const outputDir = path.join(testDir + "/OutputFile");
+
+fs.mkdir(outputDir, { recursive: true }, (err) => {
+  if (err) {
+    return console.error(err);
+  }
+});
+
+function getFileName(i: number) {
+  const source = path.join(testDir + "/SourceFile/unFormatted" + i + ".lsp");
+  const output = path.join(
+    testDir + "/OutputFile/formatedOutputFile" + i + ".lsp"
+  );
+  const baseline = path.join(
+    testDir + "/Baseline/formatedBasefile" + i + ".lsp"
+  );
+  return [source, output, baseline];
+}
+function comparefileSync(
+  i: number,
+  output: string,
+  fmt: string,
+  baseline: string
+) {
+  try {
+    fs.writeFileSync(output, fmt);
+    let baseString = fs.readFileSync(baseline, { encoding: "utf8", flag: "r" });
+    //Trick to pass the test is to ignore the \r
+    fmt = fmt.replace(/(\r)/gm, "");
+    baseString = baseString.replace(/(\r)/gm, "");
+    assert.isTrue(fmt === baseString);
+  } catch (err) {
+    assert.fail(`Format Test Case ${i} failed!`);
+  }
+}
+// Notes:
+// Format test is a setting sensitive which depends on the format settings defined
+// in the fmtconfig.ts
+// The baseline is generated by the above default value:
+// MaxLineChars: 85
+// NarrowStyleIndent: 2
+// CloseParenthesisStyle: 'New line with outer indentation'
+// LongListFormatStyle: 'Fill to margin'
+// Need to remove the \r to do the format output compare
+suite("Lisp Formatter mock Tests", function () {
+  let closeParenStyleStub;
+  let maximumLineCharsStub;
+  let longListFormatStyleStub;
+  let indentSpacesStub;
+  let internalLispFuncsStub;
+  let internalOperators;
+
+  setup(() => {
+    let keyFile = path.join(
+      __dirname + "/../../../extension/data/alllispkeys.txt"
+    );
+    internalOperators = fs.readFileSync(keyFile).toString().split("\r\n");
+    internalLispFuncsStub = ImportMock.mockOther(
+      resources,
+      "internalLispFuncs",
+      internalOperators
+    );
+  });
+  beforeEach(async () => {
+    closeParenStyleStub.restore();
+    maximumLineCharsStub.restore();
+    longListFormatStyleStub.restore();
+    indentSpacesStub.restore();
+  });
+  teardown(() => {
+    internalLispFuncsStub.restore();
+  });
+
+  suiteTeardown(async () => {
+    closeParenStyleStub.restore();
+    maximumLineCharsStub.restore();
+    longListFormatStyleStub.restore();
+    indentSpacesStub.restore();
+  });
+  suiteSetup(async () => {
+    closeParenStyleStub = ImportMock.mockFunction(
```

Comment:
you can just call setClosedParenInSameLine("...") here.
Same as other places. 
We need to minimize duplicated code.

Reply:
Good point.